plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TransactionLogEntries.java

```diff
@@ -0,0 +1,178 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.transactionlog.checkpoint;
+
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.TrinoInputStream;
+import io.trino.plugin.deltalake.transactionlog.DeltaLakeTransactionLogEntry;
+import io.trino.plugin.deltalake.transactionlog.MetadataEntry;
+import io.trino.plugin.deltalake.transactionlog.ProtocolEntry;
+import io.trino.spi.TrinoException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Streams.stream;
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+import static io.airlift.slice.SizeOf.estimatedSizeOf;
+import static io.airlift.slice.SizeOf.instanceSize;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogParser.parseJson;
+import static io.trino.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public final class TransactionLogEntries
+{
+    private static final int INSTANCE_SIZE = instanceSize(TransactionLogEntries.class);
+    private static final int JSON_LOG_ENTRY_READ_BUFFER_SIZE = 1024 * 1024;
+
+    private final long entryNumber;
+    private final Location transactionLogFilePath;
+
+    private final Optional<List<DeltaLakeTransactionLogEntry>> cachedEntries;
+
+    public TransactionLogEntries(long entryNumber, TrinoInputFile inputFile, DataSize maxCachedFileSize)
+    {
+        this.entryNumber = entryNumber;
+        this.transactionLogFilePath = inputFile.location();
+        try {
+            if (inputFile.length() > maxCachedFileSize.toBytes()) {
+                this.cachedEntries = Optional.empty();
+            }
+            else {
+                this.cachedEntries = Optional.of(ImmutableList.copyOf(new TransactionLogEntryIterator(entryNumber, inputFile)));
+            }
+        }
+        catch (IOException e) {
+            throw new TrinoException(GENERIC_INTERNAL_ERROR, "Error while reading from transaction entry iterator for the file %s".formatted(transactionLogFilePath));
+        }
+    }
+
+    /**
+     * Returns a stream of DeltaLakeTransactionLogEntry
+     * Caller has the responsibility to close this stream as it potentially holds an open file
+     */
+    public Stream<DeltaLakeTransactionLogEntry> getEntries(TrinoFileSystem fileSystem)
```

Comment:
this would likely make the interface of `DeltaLakeCommitSummary` cleaner.

Reply:
TransactionLogEntries ends up being a part of TableSnapshot cache which is used across queries. So I don't think it would be a good idea to hold on to inputFile created from one query's context to other queries.